libraries/redcore/api/hal/hal.php

```diff
@@ -1376,16 +1388,19 @@ public function checkRequiredFields($data, $configuration)
 				{
 					if (is_null($data[(string) $field['name']]) || $data[(string) $field['name']] == '')
 					{
-						JFactory::getApplication()->enqueueMessage(
-							JText::sprintf('LIB_REDCORE_API_HAL_WEBSERVICE_ERROR_REQUIRED_FIELD', (string) $field['name']), 'error'
-						);
-
-						return false;
+						$errors[] = JText::sprintf('LIB_REDCORE_API_HAL_WEBSERVICE_ERROR_REQUIRED_FIELD', (string) $field['name']);
 					}
 				}
 			}
 		}
 
+		if (!empty($errors))
+		{
+			$this->apiErrors = array_merge($this->apiErrors, $errors);
```

Comment:
ok, but where exactly are you using this variable? $this->apiErrors
It is being set, but not provided for Hal or SOAP.

And while this is good method, it should be placed on all status code 400 and above